### ISSUE TITLE
Allow AudioStreamPlayer2D to have no distance falloff like AudioStreamPlayer3D with max_distance 0

### ISSUE THREAD

**Issue by lostminds** [godotengine/godot-proposals#7109]
### Describe the project you are working on

A small scale 3d game with 2d UI sounds

### Describe the problem or limitation you are having in your project

Both AudioStreamPlayer3D and AudioStreamPlayer2D have a max_distance property that can be used for fading out the sound as the distance to the player increases. For AudioStreamPlayer3D this can be set to 0 (default) which turns off distance fade. However, for AudioStreamPlayer2D the max_distance cannot be set < 1 and it seems the distance fade cannot be turned off. The max_distance default value is 2000px, which means that you will get a noticeable change in level as you move the player from the center to the edges of the screen. This is a little unexpected and annoying in UI cases since it might make UI sound effect levels dependent on screen resolution. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Make AudioStreamPlayer2D consistent with the 3D version by allowing a max_distance to be set to 0 to disable distance fade, and maybe make this the default instead of 2000px.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Same as in AudioStreamPlayer3D.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Currently the workaround is just to set a very high max_distance and/or low attenuation value, but that's easy to forget and it would be easier and more consistent to just disable the distance fade all together if you don't want it.

### Is there a reason why this should be core and not an add-on in the asset library?

AudioStreamPlayer2D feels like a core component

**Comment by AThousandShips** [comment]
To solve this you can just attach the audio player to the character, that's the normal approach AFIK anyway if you want sounds that aren't localized, as a workaround

**Comment by lostminds** [comment]
> changing the default would break compatibility

Yeah, maybe skip that then for the first change, and just allow 0 to be set. Unless there's a way to have older version projects set the old default value and have new added nodes use the new default, which could be useful for avoiding compatibility breaks each time a default is changed.

> To solve this you can just attach the audio player to the character

As for attaching to the player, in my case there is no player and the game world camera is more or less static just zooming in and out. To me a more intuitive way of doing this is to place the UI sounds (like a meter full "ding") in the UI control components, to get the sounds in the correct 2d location on screen and bundled with the components that uses them.  The issue is that assuming the audio listener is placed in the center of the screen a 2D UI sound placed on the edge of the overlay on a 1920x1080 screen will be far enough away from the center to get a little faded with the default 2000px max distance. And since the UI overlay will resize with the window size, this distance will be different depending on the window size.

**Comment by AThousandShips** [comment]
Also can you just use `AudioStreamPlayer`? This proposal is good for various reasons imo but for UI I'd say that might be better I'm not too familiar with the audio internals

**Comment by lostminds** [comment]
> Also can you just use `AudioStreamPlayer`?

Well, in some cases I do use a AudioStreamPlayer for general UI sounds, but the benefit of the 2D version is that I can get the panning to help the player get feedback on the position of the sound in the UI. As in the "ding" coming from the meter on the right or the left, or the click sound from the on screen location of the clicked button.

**Comment by yosoyfreeman** [comment]
I think this makes sense for cohesion. I think making similar nodes work in similar ways is good.

**Comment by KoBeWi** [comment]
>  For AudioStreamPlayer3D this can be set to 0 (default) which turns off distance fade.

This doesn't seem true. Yes, 0 is default, but it does not disable fading, because the distance also depends on `unit_size`.

You can "disable" falloff by setting `max_distance` (or `unit_size` in 3D) to some very high value, like 1000000000. The property has no upper limit.

**Comment by lostminds** [comment]
> This doesn't seem true. Yes, 0 is default, but it does not disable fading, because the distance also depends on `unit_size`.

I see, I misunderstood this then based on the [AudioStreamPlayer3D documentation description for max_distance](https://docs.godotengine.org/en/stable/classes/class_audiostreamplayer3d.html#class-audiostreamplayer3d-property-max-distance). If this is reworded, perhaps a hint could be added here and in the corresponding AudioStreamPlayer2D docs to set max_distance to a very high value if you want to disable the falloff effect, as this might not be immediately obvious.

Either way I think it would be a good improvement to have a clear and easy to understand way to disable distance falloff completely both in AudioSteamPlayer2D and AudioStreamPlayer3D for cases where it's not needed or desired. If setting max_distance to 0 isn't an established way to do this already, perhaps a better way would be to introduce a new boolean `use_distance_falloff` that is true by default to keep compatibility, but when set to false bypasses all distance falloff calculation regardless of attenuation, max_distance or unit_size settings.
